.evergreen/template-yml.js

```diff
@@ -0,0 +1,87 @@
+#! /usr/bin/env node
+'use strict';
+
+const fs = require('fs');
+
+
+function template(input) {
+  // PHP-like syntax, except it's JS.
+  // foo: <% for (let i = 0; i < 10; i++) { %>
+  //   - index: <% out(i) %>
+  // <% } %>
+  // is turned into a YAML list with index: 0, index: 1 etc.
+
+  let asJs = `(function() { let result = "";
+    const out = (val) => result += (typeof val === 'object' && val !== null) ? JSON.stringify(val) : val;`;
+  let i = 0;
+  while (i < input.length) {
+    const opening = input.indexOf('<%', i);
+    if (opening === -1) {
+      asJs += `out(${JSON.stringify(input.slice(i))});\n`;
+      break;
+    }
+    asJs += `out(${JSON.stringify(input.slice(i, opening))});\n`;
+    const closing = input.indexOf('%>', opening + 2);
+    if (closing === -1) break;
+    asJs += input.slice(opening + 2, closing) + ';\n';
+
+    i = closing + 2;
+  }
+  asJs += '; return result; })()'
+  return eval(asJs);
+}
+
+
+const testPackagedAppVariations = [
+  {
+    name: 'test-packaged-app-4',
+    'test-packaged-app': {
+      vars: {
+        mongodb_version: 4
+      }
+    }
+  },
+  {
+    name: 'test-packaged-app-5',
+    'test-packaged-app': {
+      vars: {
+        mongodb_version: 5
+      }
+    }
+  }
+];
```

Comment:
I think last time we discussed this we decided to have the matrix matching the mongosh setup except for the alpha release. I think we can also initially include only either enterprise or community just to not overwhelm the CI with tasks, but I would add all the different 4.x and 5 right away:

```suggestion
const testPackagedAppVariations = [
  {
    name: 'test-packaged-app-40x',
    'test-packaged-app': {
      vars: {
        mongodb_version: '4.0.x'
      }
    }
  },
  {
    name: 'test-packaged-app-42x',
    'test-packaged-app': {
      vars: {
        mongodb_version: '4.2.x'
      }
    }
  },
  {
    name: 'test-packaged-app-44x',
    'test-packaged-app': {
      vars: {
        mongodb_version: '4.4.x'
      }
    }
  },
  {
    name: 'test-packaged-app-5x',
    'test-packaged-app': {
      vars: {
        mongodb_version: '5.x'
      }
    }
  }
];
```

I am not sure actually what's the best strategy for adding versions that were released under the "rapid release" policy, lock it to 5.x to match Atlas rapid release updates, have all the rapid release patches separately, or do a 5.0.x as the one that is the latest version in the download center /cc @mmarcon @addaleax maybe you have some thoughts to share here)

And here's mongosh for reference (totally feel free to ignore my suggestion above and add these in any way you find better matching your code style)

https://github.com/mongodb-js/mongosh/blob/main/.evergreen/evergreen.yml.in#L8-L18

Reply:
Well, I think the main reason why it’s just fine to skip the rapid releases in mongosh is that we also check the latest alpha versions, so we catch server-side changes fairly early. If we only checked 5.0.x, 6.0.x, etc. in mongosh, we’d run the risk of not being aware of server changes until the next major version is released, i.e. possibly a year later than we’d otherwise find out. This is probably less of an issue for Compass, but it’s something to keep in mind.

I would ideally, if we can, like to test both enterprise and community servers. I don’t think that comes with much overhead for us here?

Reply:
Yeah, I think it should be fine to do both enterprise and community, I was being overly cautious about resources this will require when suggesting this. It might be a bottleneck when we get macos machines (as we already know that even when we get them, there will be a limited amount), but as we don't run any e2e tests on mac right now, it's not an issue 👍

Reply:
I've committed this so long, btw.